quadriga/src/main/java/edu/asu/diging/quadriga/core/conceptpower/service/impl/ConceptPowerServiceImpl.java

```diff
@@ -0,0 +1,207 @@
+       
+        return updated ? conceptCache : conceptCacheOld;
+    }
+
+    /**
+     * Updates Concept Type
+     * @param conceptCache
+     * @param conceptCacheOld
+     * @return
+     */
+    private boolean updateConceptType(ConceptCache conceptCache, ConceptCache conceptCacheOld) {
+        if(conceptCache != null && conceptCache.getConceptType() != null) {
+
+            ConceptType conceptType = conceptCache.getConceptType();
+            ConceptType conceptTypeOld = null;
+
+            if(conceptCacheOld != null) {
+                conceptTypeOld = conceptCacheOld.getConceptType();
+            }
+
+            // ConceptPower returned a concept type and either no conceptType entry exists in the DB
+            // or if one exists, it is different from the current conceptType entry
+            if (conceptType != null && (conceptTypeOld == null || conceptTypeOld.compareTo(conceptType) != 0)) {
+               
+                conceptCache.setConceptType(conceptType);
+                conceptTypeService.saveConceptType(conceptType);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Updates Concept cache
+     * @param conceptCache
+     * @param conceptCacheOld
+     * @param uri
+     * @return
+     */
+    private boolean updateConceptCache(ConceptCache conceptCache, ConceptCache conceptCacheOld, String uri) {
+        
+        // ConceptPower returned a concept and either no conceptCache entry exists in the DB
+        // or if one exists, it is different from the current conceptCache entry
+        if (conceptCache != null && (conceptCacheOld == null || conceptCacheOld.compareTo(conceptCache) != 0)) {
+            
+           
+            
+            conceptCacheService.saveConceptCache(conceptCache);
+            
+            Optional.ofNullable(conceptCache.getAlternativeUris())
+                    .ifPresent(nonNullAlternativeUris -> nonNullAlternativeUris
+                            .stream()
+                            .filter(alternativeUri -> alternativeUri != null)
+                            .filter(nonNullAlternativeUri -> !nonNullAlternativeUri.equals(conceptCache.getUri()))
+                            .forEach(alternativeUriValue -> conceptCacheService.deleteConceptCacheByUri(alternativeUriValue)));  
+            return true;
+            
+        } else if(conceptCache == null) {
+            logger.error("ConceptPower did not return any concept entries for uri: " + uri);
+           
+        }
+        return false;
+    }
+
+    @Override
+    public ConceptCache mapConceptPowerReplyToConceptCache(ConceptPowerReply conceptPowerReply) {
+        // If we get multiple ConceptPower entries in the reply, we use the first one
+        List<ConceptEntry> conceptEntries = conceptPowerReply.getConceptEntries();
+        ConceptCache conceptCache = null;
+
+        if (conceptEntries != null && !conceptEntries.isEmpty()) {
```

Comment:
turn this if around and just return if this condition is not met. It'll unindent all the following code.